### PR TITLE
Added options

### PR DESCRIPTION
**Pull request by kimbo**
@JustBrenkman requesting your review.

Added options for:
- Compiling before running test cases
- Testing for complexity
- Testing for memory leaks

Updated README.md with explanation of what the options are.

Minor changes:
- Added ignore all whitespace flag to diff.
- Added shebang line at the top

### CODE REVIEW

File: testbench.sh [hunk 1]
```diff
@@ -36,6 +69,43 @@ echo
...
+  if [ $? != 0 ]; then
+    echo -e "  ${RED}ERROR: could not complete complexity test. Exiting...${NC}";
+    echo "${complexity_output}"
+    exit
+  fi
+
+  errors=$(awk '($1>8)' <<< ${complexity_output})
+
+  if [ -n "${errors}" ]; then
+    printf "%s:%schicken\n" "Errors" ${errors}
```
Comment (kimbo) [inline testbench.sh]: Oops. Forgot to remove this line hehe :laughing:

Comment (kimbo) [reply testbench.sh]: I'll fix that tomorrow

Comment (kimbo) [reply testbench.sh]: Or now

**Review by JustBrenkman — APPROVED** [review]
Great additions, you should also add an option to test all options. Kinda like the real pass off.

**Comment by kimbo** [comment]
@JustBrenkman I had the same thought lol.

**Pull request merged**